storage-dynamodb/integration_tests/dynamodb_logstore.py

```diff
@@ -90,24 +81,41 @@
     .master("local[*]") \
     .config("spark.sql.extensions", "io.delta.sql.DeltaSparkSessionExtension") \
     .config("spark.delta.logStore.class", delta_storage) \
-    .config("spark.delta.DynamoDBLogStoreScala.tableName", dynamo_table_name) \
-    .config("spark.delta.DynamoDBLogStoreScala.region", dynamo_region) \
-    .config("spark.delta.DynamoDBLogStoreScala.errorRates", dynamo_error_rates) \
+    .config("spark.delta.DynamoDBLogStore.tableName", dynamo_table_name) \
+    .config("spark.delta.DynamoDBLogStore.region", dynamo_region) \
+    .config("spark.delta.DynamoDBLogStore.errorRates", dynamo_error_rates) \
     .getOrCreate()
 
-data = spark.createDataFrame([], "id: int, a: int")
-data.write.format("delta").mode("overwrite").partitionBy("id").save(delta_table_path)
+SCHEMA = "run_id: string, id: int, a: int"
+
+RUN_ID = str(uuid.uuid4())
+
+data = spark.createDataFrame([], SCHEMA)
+
+if table_overwrite:
+    data.write.format("delta").mode("overwrite").partitionBy("run_id", "id").save(delta_table_path)
```

Comment:
should we also empty the DDB table?

Reply:
It is not necessary, if we want to start with no dynamodb entries for table we can create table in new location.

Reply:
But maybe we can simply remove this whole step? The intention was to create empty table before starting multi-thread writing, but maybe it is not necessary. I'll test it

Reply:
Ok, something interesting happened when I ran this test without this 'table creation' step:
```
delta.exceptions.ProtocolChangedException: The protocol version of the Delta table has been changed by a concurrent update. This happens when multiple writers are writing to an empty directory. Creating the table ahead of time will avoid this conflict. Please try the operation again.
Conflicting commit: {"timestamp":1648832395501,"operation":"WRITE","operationParameters":{"mode":Append,"partitionBy":["run_id","id"]},"isolationLevel":"Serializable","isBlindAppend":true,"operationMetrics":{"numFiles":"1","numOutputRows":"1","numOutputBytes":"449"},"engineInfo":"Apache-Spark/3.2.0 Delta-Lake/1.2.0-SNAPSHOT","txnId":"bcd0f41e-d9ff-4455-ad02-e18d57c150d2"}
```
The interesting thing it failed with above message after inserting ~50 row to this table (I would expect quick fail with no rows inserted at all). Probably single thread failed after some timeout while other threads were inserting data.

Reply:
Ok - I simplified whole thing by removing this conditional overwrite. Instead table is created if necessary by appending  empty data unconditionally.